rand_distr/src/triangular.rs

```diff
@@ -44,13 +44,13 @@ impl Triangular {
     /// `mode`.
     #[inline]
     pub fn new(min: f64, max: f64, mode: f64) -> Result<Triangular, Error> {
-        if max < mode {
+        if !(max >= mode) {
             return Err(Error::MaxTooSmall);
         }
-        if mode < min {
+        if !(mode >= min) {
             return Err(Error::ModeTooSmall);
         }
-        if max == min {
+        if !(max != min) {
```

Comment:
In this case I don't think the double-negation does anything different than `==`. It doesn't matter however since the above lines already rule out either being NaN.

Reply:
That's true. I guess it is more consistent with the other conditions? I don't really prefer either way.